avoid upstream scoping issues

jdlrobson opened the pull request:
create an encoded admin tree from scratch
more fixtures and tests

jrobson committed:
avoid upstream scoping issues

create an encoded admin tree from scratch
more fixtures and tests

awjrichards-zz commented:
With this changeset I am still seeing encoding problems - this is the admtree string that I see in the API (taken from my log output):
Admin tree string: us|us-ca|%5B%5BAlameda%20County%2C%20California%5D%5D

I believe there is still an encoding problem in the app. For instance, if I visit:
http://wlm.wikimedia.org/api/api.php?action=adminlevels&uselang=en&admtree=us|us-ca|%5B%5BAlameda%20County%2C%20California%5D%5D
The string gets properly decoded. My log output looks like:
Admin tree string: us|us-ca|[[Alameda County, California]]

jrobson committed:
fix issue related to back button

arthur pointed out a nasty recursion bug related to encoding issues.
this tests describes it
update fixtures to not url encode (jquery seems to do this for us)
when going back make sure we decode before listing campaigns as otherwise
we will request a different campaign (one with a url encoded name)
this seems to confuse app

jrobson committed:
escape pipe symbols in campaign names

some of the names have pipes in eek which is the same as our
separate. we escape these so as not to confuse the api
